src/team_lib/mhba/xccl_mhba_collective.c

```diff
@@ -537,25 +750,13 @@ xccl_status_t xccl_mhba_alltoall_init(xccl_coll_op_args_t  *coll_args,
     block_size = team->requested_block_size ? team->requested_block_size : block_size;
 
     //todo following section correct assuming homogenous PPN across all nodes
-    if (team->node.sbgp->group_size % block_size != 0) {
-        if (team->node.sbgp->group_rank == team->node.asr_rank) {
-            xccl_mhba_warn("Block size was decreased to fit node PPN");
-        }
-        while (team->node.sbgp->group_size % block_size && block_size > 2) {
-            block_size -= 1;
-        }
-    }
+    request->num_of_blocks_columns = (team->node.sbgp->group_size % block_size) ?
+            xccl_round_up(team->node.sbgp->group_size,block_size) : 0;
     block_msgsize = SQUARED(block_size) * len;
-    if (team->node.sbgp->group_rank == team->node.asr_rank) {
-        xccl_mhba_info("Block size is %d", block_size);
-    }
-    if (team->node.sbgp->group_size % block_size) {
-        if (team->node.sbgp->group_rank == team->node.asr_rank) {
-            xccl_mhba_error("node PPN can't be divided by any block size, or "
-                            "block size is 1 - NOT SUPPORTED");
-        }
-        return XCCL_ERR_NO_RESOURCE;
+    if (((team->net.sbgp->group_rank == 0) && (team->node.sbgp->group_rank == team->node.asr_rank)) && (len != team->previous_msg_size[request->seq_index])) {
```

Comment:
line too long, plz adhere to code style

Reply:
do we need this conditional at all? only for info ?

Reply:
without it we have too much messages.... i used it a lot in my debug, but i think it applies for production as well